outlines/fsm/guide.py

```diff
@@ -193,12 +193,8 @@ def get_next_state(self, state: int, token_id: int) -> int:
         The new state of the guide.
 
         """
-        if token_id == self.eos_token_id:
+        if token_id == self.eos_token_id or state not in self.states_to_token_maps:
```

Comment:
Shouldn;t `state` be always present in `self.states_to_token_maps`? If not then we should throw an error or letting it pass silently?